scripts/check-docstrings.py

```
@@ -0,0 +1,126 @@
+    """Every undocumented item clippy can see, as (file, line, what)."""
+    result = subprocess.run(
+        [
+            "cargo", "clippy", "--all-targets", "--message-format=json",
+            "--quiet", "--", "-W", LINT,
+        ],
+        capture_output=True,
+        text=True,
+    )
+    if result.returncode != 0 and not result.stdout:
+        print(result.stderr, file=sys.stderr)
+        raise SystemExit("cargo clippy failed; fix the build first")
```

Ruff (0.16.3)

[warning] Avoid specifying long messages outside the exception class

(TRY003)